Formula/n/node.rb

```diff
@@ -33,14 +33,24 @@ class Node < Formula
   uses_from_macos "zlib"
 
   on_macos do
-    depends_on "llvm" => [:build, :test] if DevelopmentTools.clang_build_version <= 1100
+    depends_on "llvm" => :build if DevelopmentTools.clang_build_version <= 1500
   end
 
+  on_linux do
+    depends_on "llvm" => :build if DevelopmentTools.gcc_version("/usr/bin/gcc") < 12
```

Comment:
I guess `node` is mainly for the runtime and doesn't have shared library issue so could try `gcc`:
```suggestion
    depends_on "gcc" if DevelopmentTools.gcc_version("/usr/bin/gcc") < 12
```